addons/sale_mrp/models/sale_mrp.py

```diff
@@ -11,19 +11,35 @@ class SaleOrderLine(models.Model):
     @api.multi
     def _compute_qty_delivered(self):
         super(SaleOrderLine, self)._compute_qty_delivered()
-
-        for line in self:
-            if line.qty_delivered_method == 'stock_move':
-                # In the case of a kit, we need to check if all components are shipped. Since the BOM might
-                # have changed, we don't compute the quantities but verify the move state.
-                bom = self.env['mrp.bom']._bom_find(product=line.product_id)
-                if bom and bom.type == 'phantom':
-                    moves = line.move_ids.filtered(lambda m: m.picking_id and m.picking_id.state != 'cancel')
-                    bom_delivered = all([move.state == 'done' for move in moves])
-                    if bom_delivered:
-                        line.qty_delivered = line.product_uom_qty
+        for order_line in self:
+            if order_line.qty_delivered_method == 'stock_move':
+                bom = self.env['mrp.bom']._bom_find(product=order_line.product_id, bom_type='phantom')
+                if bom:
+                    moves = order_line.move_ids.filtered(lambda m: m.state == 'done' and not m.scrapped)
+                    qty_ratios = []
+                    order_uom_qty = order_line.product_uom._compute_quantity(order_line.product_uom_qty, bom.product_uom_id)
+                    boms, bom_sub_lines = bom.explode(order_line.product_id, order_uom_qty)
+                    for bs_line in bom_sub_lines:
+                        bom_line = bs_line[0]
+                        bom_line_data = bs_line[1]
+                        if bom_line.product_id in moves.mapped('product_id'):
+                            relevant_moves = moves.filtered(lambda m: m.bom_line_id == bom_line)
```

Comment:
i don't like it
we should probably use bom_line_id on the stock.move and fill it if itsn't the case with kit at the moment